packages/material-ui/src/Paper/Paper.js

```diff
@@ -38,9 +37,7 @@ const Paper = React.forwardRef(function Paper(props, ref) {
   } = props;
 
   if (process.env.NODE_ENV !== 'production') {
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    const theme = useTheme();
-    if (!theme.shadows[elevation]) {
+    if (classes[`elevation${elevation}`] === undefined) {
```

Comment:
We could technically move this into propTypes but I wanted to avoid adding hurdles when migrating from withStyles to makeStyles.

Nice side-effect is that styles creation and this warning are now decoupled. Previously refactoring of how we would create the elevation classes would require refactoring this check.